src/components/MessagesList/MessagesList.vue

```diff
@@ -268,15 +286,32 @@ export default {
 		token(newToken, oldToken) {
 			// Expire older messages when navigating to another conversation
 			this.$store.dispatch('easeMessageList', { token: oldToken })
+			this.messagesGroupedByDateByAuthor = this.prepareMessagesGroups(this.messagesList)
 		},
 
 		messagesList: {
 			immediate: true,
-			handler(messages) {
-				const groups = this.prepareMessagesGroups(messages)
-				this.messagesGroupedByAuthor = this.softUpdateMessagesGroups(this.messagesGroupedByAuthor, groups)
+			handler(newMessages, oldMessages) {
+				// token watcher will handle the conversations change
+				if (oldMessages?.length && newMessages.length && newMessages[0].token !== oldMessages?.at(0)?.token) {
+					return
+				}
+				const newGroups = this.prepareMessagesGroups(newMessages)
+				// messages were just loaded
+				if (!oldMessages) {
+					this.messagesGroupedByDateByAuthor = newGroups
+				} else {
+					this.softUpdateByDateGroups(this.messagesGroupedByDateByAuthor, newGroups)
+				}
 			},
 		},
+
+		currentDay() {
+			this.dateSeparatorLabels = {}
```

Comment:
> only when token / messagesList changes

It falls to the same problem, we need date seps to be updated without any changes needed to trigger the update.

>  5 min interval

I tried to avoid this :(

Reply:
then let's revert to using method in template, and put a follow-up issue?